src/main/java/beam/analysis/plots/filterevent/ActivitySimFilterEvent.scala

```diff
@@ -0,0 +1,47 @@
+package beam.analysis.plots.filterevent
+
+import scala.collection.JavaConverters._
+
+import beam.agentsim.events.{ModeChoiceEvent, ReplanningEvent}
+import beam.sim.config.BeamConfig
+import org.matsim.api.core.v01.events.Event
+import org.matsim.api.core.v01.population.{Activity, PlanElement}
+import org.matsim.core.controler.MatsimServices
+
+class ActivitySimFilterEvent(beamConfig: BeamConfig, matsimServices: MatsimServices) extends FilterEvent {
+
+  override def graphNamePreSuffix: String = "_commute"
+
+  override def shouldProcessEvent(event: Event): Boolean = {
+    val isEnabled = beamConfig.beam.exchange.scenario.urbansim.activitySimEnabled
+    event match {
+      case mcEvent: ModeChoiceEvent if isEnabled => isHomeOrWorkActivity(mcEvent)
+      case rEvent: ReplanningEvent if isEnabled  => isHomeOrWorkActivity(rEvent)
+      case _                                     => false
+    }
+  }
+
+  private def isHomeOrWorkActivity(event: ModeChoiceEvent): Boolean = {
+    val person = matsimServices.getScenario.getPopulation.getPersons.get(event.personId)
+    val plan = person.getSelectedPlan
+    val planElements = person.getSelectedPlan.getPlanElements.asScala.toIndexedSeq
+    val current: Option[PlanElement] = planElements.lift(event.tourIndex - 1)
+    val next: Option[PlanElement] = planElements.lift(event.tourIndex)
+    isHomeOrWorkActivity(current) && isHomeOrWorkActivity(next)
+  }
+
+  private def isHomeOrWorkActivity(event: ReplanningEvent): Boolean = {
+    isHomeOrWorkActivity(event.getEventType)
+  }
+
+  private def isHomeOrWorkActivity(activity: Option[PlanElement]): Boolean = {
```

Comment:
nitpick: pattern matching can be replaces by `Option[PlanElement].exists()` method

Reply:
As per discussion, you're doing pattern-matching to get the Activity type, `exists` won't work here, so no need to replace it.